core/src/main/scala/kafka/coordinator/group/GroupMetadataManager.scala

```diff
@@ -39,7 +39,7 @@ import org.apache.kafka.common.protocol.Errors
 import org.apache.kafka.common.protocol.types.Type._
 import org.apache.kafka.common.protocol.types._
 import org.apache.kafka.common.record._
-import org.apache.kafka.common.requests.{IsolationLevel, OffsetFetchResponse}
+import org.apache.kafka.common.requests.{IsolationLevel, OffsetCommitRequest, OffsetFetchResponse}
```

Comment:
Putting this comment here for lack of a better place. We need to update `storeGroup` to check the inter-broker format before using the new version of the group metadata schema. I am puzzled why we don't have a similar check for `offsetCommitValue`. I suspect it is a bug. We could fix it here or in a follow-up.

Reply:
Thanks for pointing this out. I fixed it in the new commit. Regarding `offsetCommitValue`, wouldn't the existing check in `offsetCommitValue(...)` handle the expected version detection?

    val offsetValueSchemaVersion: Short =
      // if expire timestamp is explicitly provided use the old schema
      if (offsetAndMetadata.expireTimestamp.nonEmpty) 1
      //  otherwise, use the current v2 schema
      else 2